src/Workspaces/Core/Portable/Options/GlobalOptionService.cs

```diff
@@ -99,6 +103,47 @@ public IEnumerable<IOption> GetRegisteredOptions()
             return _lazyAllOptions.Value;
         }
 
+        public bool TryMapEditorConfigKeyToOption(string key, string? language, [NotNullWhen(true)] out IEditorConfigStorageLocation2? storageLocation, out OptionKey optionKey)
+        {
+            var (option, storage) = ImmutableInterlocked.GetOrAdd(
+                ref _editorConfigKeysToOptions,
+                key,
+                (key, self) => MapToOptionWithoutLanguage(self, key),
+                this);
+
+            if (option is object)
+            {
+                RoslynDebug.AssertNotNull(storage);
+                storageLocation = storage;
+                optionKey = option.IsPerLanguage ? new OptionKey(option, language) : new OptionKey(option);
+                return true;
+            }
+
+            storageLocation = null;
+            optionKey = default;
+            return false;
+
+            // Local function
+            static (IOption? option, IEditorConfigStorageLocation2? storageLocation) MapToOptionWithoutLanguage(GlobalOptionService service, string key)
+            {
+                foreach (var option in service.GetRegisteredOptions())
```

Comment:
`GetRegisteredOptions` will force load all the assemblies which have any `IOptionProvider`. You should instead use `GetRegisteredSerializableOptions(language)` to only get options for specific language + language agnostic options. This will not load C# assemblies from VB only solution and vice versa. All our editorconfig based options are serializable and part of serializable option set.

Reply:
It's a little weird now, but I implemented it with `GetRegisteredSerializableOptions` without losing the cache(s) for the three common language cases (C#, Visual Basic, unspecified).